openmc/deplete/results_list.py

```diff
@@ -192,3 +193,72 @@ def get_depletion_time(self):
         for ix, res in enumerate(items):
             times[ix] = res.proc_time
         return times
+
+    def export_to_materials_xml(self, burnup_index, nuc_with_data=None, new_xml_name=None):
+        """Export Materials.xml file from the depletion results
+
+        Parameters
+        ----------
+        burn_index : int
+            Index of burnup step to evaluate
+        nuc_with_data : Iterable of str, optional
+            nuclides list to evaluate with neutron data. If not provided,
+            nuclides from the cross_sections element of materials.xml will
+            be used. If that element is not present, nuclides from
+            OPENMC_CROSS_SECTIONS will be used.
+        new_xml_name : str, optional
+            name of xml file to put depleted materials into. If not
+            provided, no new xml file will be written.
+
+        Returns
+        -------
+        mat_file : Materials
+            A modified Materials instance containing depleted material data,
+            plus original isotopic compositions of non-depletable materials
+        """
+        result = self[burnup_index]
+
+        # Only materials found in the original materials.xml file will be
+        # updated. If for some reason you have modified OpenMC to produce
+        # new materials as depletion takes place, this method will not
+        # work as expected and leave out that material.
+        mat_file = openmc.Materials.from_xml("materials.xml")
+
+        # Only nuclides with valid transport data will be written to
+        # the new materials XML file. The precedence of nuclides to select
+        # is first ones provided as a kwarg here, then ones specified
+        # in the materials.xml file if provided, then finally from
+        # the environment variable OPENMC_CROSS_SECTIONS.
+        if nuc_with_data:
+            available_cross_sections = nuc_with_data
```

Comment:
Alright, sounds good. I think it's best to avoid set conversion though. If the user passes in a set, great. This code is in no way performance-critical or slow from the get-go, so I see no point in converting to a set. It just adds extra code. May as well keep it minimal. What if the user passes in some iterable which isn't convertible to a set? Unlikely, yes, but I'd rather not convert for those two reasons.

Reply:
Looking at some timing comparisons, finding something in a set is about 10x faster than finding something in a list. But we also take a noticeable hit in the list (or other iterable) to set conversion. So here it doesn't make as much sense as below when we're making the set by hand. (which you pointed out below too)